test/test_unary_ufuncs.py

```diff
@@ -70,10 +70,17 @@
 _large_size = (1029, 917)
 
 
+# Replace values satisfying condition with a safe value. This is used to block
+# out values the could cause singularity like tan(pi/2)
+def replace_values_in_tensor(tensor, condition, safe_value):
+    mask = condition(tensor).to(tensor.device)
```

Comment:
If this call is guarded then I think it's OK to assume that this `to()` is unnecessary. That is, we can require the mask generator return a tensor on the same device as the input